flow/Arena.cpp

```diff
@@ -20,8 +20,93 @@
 
 #include "Arena.h"
 
+// See https://dox.ipxe.org/memcheck_8h_source.html and https://dox.ipxe.org/valgrind_8h_source.html for an explanation
+// of valgrind client requests
+#ifdef VALGRIND_ARENA
+#include <memcheck.h>
+#else
+#define VALGRIND_MAKE_MEM_NOACCESS(addr, size) ((void)(addr), (void)(size))
+#define VALGRIND_MAKE_MEM_DEFINED(addr, size) ((void)(addr), (void)(size))
+#define VALGRIND_MAKE_MEM_UNDEFINED(addr, size) ((void)(addr), (void)(size))
+#endif
+
+// For each use of arena-internal memory (e.g. ArenaBlock::getSize()), unpoison the memory before use and
+// poison it when done.
+// When creating a new ArenaBlock, poison the memory that will be later allocated to users.
+// When allocating memory to a user, mark that memory as undefined.
+
+namespace {
+void unpoison(ArenaBlock* b) {
```

Comment:
nit: `unpoison` and `poison` confused me on my first read.  When I think of poisoning memory, I think of writing a special well known pattern (to hopefully induce a crash on an assert or bad pointer dereference).  Maybe rename these to `allow_access()`, `disallow_access()`, or something?

Reply:
Done